morpheus/pipeline/single_port_stage.py

```diff
@@ -64,6 +66,14 @@ def _pre_compute_schema(self, schema: _pipeline.StageSchema):
         if (not typing_utils.issubtype(input_type, accepted_types)):
             raise RuntimeError((f"The {self.name} stage cannot handle input of {input_type}. "
                                 f"Accepted input types: {self.accepted_types()}"))
+        out_type = MultiResponseMessage
+        if (schema.input_type == ControlMessage):
+            self._use_control_message = True
+            out_type = ControlMessage
+        else:
+            self._use_control_message = False
+
+        schema.output_schema.set_type(out_type)
```

Comment:
This is a bit too general but I like the direction. Not all return types for every class that derives from `SinglePortStage` will return a `MultiResponseMessage` or a `ControlMessage`. But we could save off the input/output type to make it easier to use in derived classes. Something like this would work:

```suggestion
        self._input_type = schema.input_type
```

And then you can just use `if (self._input_type == ControlMessage)` in derived classes